modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/ComponentBasedValueProviderCacheIdGenerator.java

```diff
@@ -6,62 +6,33 @@
  */
 package org.mule.runtime.config.api.dsl.model.metadata;
 
-import static java.util.Optional.empty;
 import static org.mule.runtime.api.util.Preconditions.checkArgument;
+
 import org.mule.runtime.api.dsl.DslResolvingContext;
-import org.mule.runtime.api.meta.model.connection.ConnectionProviderModel;
 import org.mule.runtime.ast.api.ComponentAst;
-import org.mule.runtime.config.api.dsl.model.DslElementModel;
-import org.mule.runtime.config.api.dsl.model.DslElementModelFactory;
-import org.mule.runtime.config.internal.model.ComponentModel;
 import org.mule.runtime.core.internal.locator.ComponentLocator;
 import org.mule.runtime.core.internal.value.cache.ValueProviderCacheId;
 import org.mule.runtime.core.internal.value.cache.ValueProviderCacheIdGenerator;
 
 import java.util.Optional;
 
+/**
+ * @deprecated Use {@link ComponentAstBasedValueProviderCacheIdGenerator} directly.
```

Comment:
Why not remove this class too?

Reply:
tooling is using it

Reply:
But we should change it so that it uses the new implementation. Al least, let's create a jira ticket and add a comment for it.